internal/searcher/client.go

```diff
@@ -55,13 +55,14 @@ func (c *client) SearchVersion(query, version string) (*SearchResult, error) {
 	)
 }
 
-func (c *client) Resolve(pkg, version string) (*lock.Package, error) {
-	result, err := c.SearchVersion(pkg, version)
+func (c *client) Resolve(pkg string) (*lock.Package, error) {
+	name, version, _ := strings.Cut(pkg, "@")
+	result, err := c.SearchVersion(name, version)
```

Comment:
Maybe we should rename it as `SearchPackageWithVersion` or something else?

Reply:
I kinda like the succinctness of `Search` and `SearchVersion`.